crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -197,8 +198,9 @@ static int is_md_fips_approved_for_verifying(int md_type, int pkey_type) {
     case NID_sha384:
     case NID_sha512:
       return 1;
+    case NID_sha512_224:
     case NID_sha512_256:
-      // SHA512/256 is only approved for verifying with RSA PSS
+      // Truncated SHA512 is only approved for verifying with RSA PSS
       if (pkey_type == EVP_PKEY_RSA_PSS) {
         return 1;
       }
```

Comment:
I'm pattern-matching to SHA512-256 here. Does anyone know of a NIST document (or similar) that stipulates which usages of these truncated hash algorithms are FIPS-approved?

Reply:
page 22 of [this NIST document](https://pages.nist.gov/ACVP/draft-celi-acvp-rsa.txt) seems to indicate that SHA512-224 is viable for RSA padding.